compiler/noirc_frontend/src/utils.rs

```diff
@@ -0,0 +1,17 @@
+use acvm::{AcirField, FieldElement};
+use num_bigint::{BigInt, BigUint, Sign};
+
+pub fn big_int_to_field_element(big_int: &BigInt) -> FieldElement {
+    let big_uint = big_int.magnitude();
+    FieldElement::from_be_bytes_reduce(&big_uint.to_bytes_be())
+}
```

Comment:
This is going to map the integer `-1` to the field element `1` when it should instead be mapped to `p-1`.

Reply:
Ah, good catch. I pushed a commit to fix this.